src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js

```diff
@@ -68,7 +76,25 @@ const DomainList = (props) => {
           </View>
         </View>
       </TouchableOpacity>
-      {item.isunfollowed ? (
+      {isBlockedSection ? (
+        item.isUnblocked ? (
+          <TouchableNativeFeedback testID="isUnblock" onPress={handleSetBlock}>
+            <View style={styles.followContainer}>
+              <View style={styles.buttonBlockUser}>
+                <Text style={styles.textButtonBlockUser}>Block</Text>
+              </View>
+            </View>
+          </TouchableNativeFeedback>
+        ) : (
+          <TouchableNativeFeedback testID="isBlock" onPress={handleSetUnblock}>
+            <View style={styles.followContainer}>
+              <View style={styles.buttonBlock}>
+                <Text style={styles.textButtonBlock}>Blocked</Text>
+              </View>
+            </View>
+          </TouchableNativeFeedback>
+        )
+      ) : item.isunfollowed ? (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block introduces conditional rendering based on the `isBlockedSection` prop and the `item.isUnblocked` state. It's good practice to handle different states in this way. However, make sure that the `handleSetBlock` and `handleSetUnblock` functions are properly defined and handle all edge cases, as they directly affect user interaction.

```diff
- {item.isunfollowed ? (
-    <TouchableNativeFeedback onPress={handleSetFollow}>
-      <View style={styles.followContainer}>
-        <View style={styles.buttonFollow}>
+ {isBlockedSection ? (
+   item.isUnblocked ? (
+     <TouchableNativeFeedback testID="isUnblock" onPress={handleSetBlock}>
+       <View style={styles.followContainer}>
+         <View style={styles.buttonBlockUser}>
+           <Text style={styles.textButtonBlockUser}>Block</Text>
+         </View>
+       </View>
+     </TouchableNativeFeedback>
+   ) : (
+     <TouchableNativeFeedback testID="isBlock" onPress={handleSetUnblock}>
+       <View style={styles.followContainer}>
+         <View style={styles.buttonBlock}>
+           <Text style={styles.textButtonBlock}>Blocked</Text>
+         </View>
+       </View>
+     </TouchableNativeFeedback>
+   )
+ ) : item.isunfollowed ? (
+   <TouchableNativeFeedback onPress={handleSetFollow}>
+     <View style={styles.followContainer}>
+       <View style={styles.buttonFollow}>
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->